src/Gelf/Transport/RetryTransportWrapper.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Gelf\Transport;
+
+use Gelf\MessageInterface as Message;
+use RuntimeException;
+
+class RetryTransportWrapper extends AbstractTransport
+{
+    /**
+     * @const string
+     */
+    const NO_RESPONSE = "Graylog-Server didn't answer properly, expected 'HTTP/1.x 202 Accepted', response is ''";
+
+    /**
+     * @var TransportInterface
+     */
+    protected $transport;
+
+    /**
+     * @var int
+     */
+    protected $maxRetries;
+
+    /**
+     * @var callable|null
+     */
+    protected $exceptionMatcher;
+
+    /**
+     * KeepAliveRetryTransportWrapper constructor.
+     *
+     * @param TransportInterface $transport
+     * @param int $maxRetries
+     * @param callable|null $exceptionMatcher
+     */
+    public function __construct(TransportInterface $transport, $maxRetries, $exceptionMatcher = null)
+    {
+        $this->transport = $transport;
+        $this->maxRetries = $maxRetries;
+        $this->exceptionMatcher = $exceptionMatcher;
+    }
+
+    /**
+     * @return TransportInterface
+     */
+    public function getTransport()
+    {
+        return $this->transport;
+    }
+
+    /**
+     * Sends a Message over this transport.
+     *
+     * @param Message $message
+     *
+     * @return int calls function to send message
+     */
+    public function send(Message $message)
+    {
+        $tries = 0;
+
+        while (true) {
+            try {
+                $tries++;
+                return $this->transport->send($message);
+            } catch (RuntimeException $e) {
```

Comment:
We should probably capture `\Throwable` here. We do not know what kind of Exceptions we can expect from a Transport

Reply:
This change did not work unless i switched the project's php version to 7.0+ Do you approve of this?

Reply:
Hi! I'm writing this follow-up to make sure this question doesn't go unnoticed. Thank you!